src/lib/utils.ts

```diff
@@ -0,0 +1,124 @@
+import type { JSONSchema, SchemaObject } from 'oas/dist/rmoas.types';
+
+import { get as lodashGet } from 'lodash'; // eslint-disable-line no-restricted-imports
+
+/**
+ * Determine if a schema `type` is, or contains, a specific discriminator.
+ *
+ */
+export function hasSchemaType(
+  schema: SchemaObject,
+  discriminator: 'array' | 'object' | 'string' | 'number' | 'boolean' | 'integer' | 'null'
+) {
+  if (Array.isArray(schema.type)) {
+    return schema.type.includes(discriminator);
+  }
+
+  return schema.type === discriminator;
+}
+
+/**
+ * With a supplied JSON Schema object, spider through it for any schemas that may contain specific
+ * kind of `format` that also happen to be within the current `requestBody` payload that we're
+ * creating a HAR representation for.
+ *
+ */
+export function getTypedFormatsInSchema(
+  format: 'json' | 'binary',
+  schema: any,
+  opts: {
+    parentIsArray?: boolean;
```

Comment:
Could this not be inferred via the `parentKey`?

Reply:
Unfortunately not because `parentKey` could be a parent key for an object, or a regular schema.